RELEASE.md

```diff
@@ -14,35 +14,63 @@
+1. Make sure your repo is clean by git's standards. It is recommended to use a fresh checkout.
+1. When bumping `X` or `Y` (but not Z or the pre-release suffix) in the release version you must create a new release branch called `release-X.Y`.
+   > NOTE: `upstream` should be the name of the remote pointing to `github.com/kubernetes-sigs/cluster-api-provider-openstack`
+    - git checkout main
+    - git pull
+    - git checkout -b release-`X`.`Y`
+    - git push --set-upstream `upstream`
+1. When bumping `X` or `Y` (but not Z or the pre-release suffix) in the release version, ensure you have added a new
+   entry to [metadata.yaml](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/metadata.yaml)
+   as [described in the CAPI book](https://cluster-api.sigs.k8s.io/clusterctl/provider-contract.html#metadata-yaml), and
+   that this has been committed to the release branch prior to release.
+1. Make sure you are on the correct release branch: `release-X.Y`
 1. Set an environment variable with the version, e.g.:
-    - `VERSION=v0.4.0`
+    - `VERSION=v0.6.0`
 1. Create an annotated tag
     - `git tag -s -a $VERSION -m $VERSION`.
 1. Push the tag to the GitHub repository:
    > NOTE: `upstream` should be the name of the remote pointing to `github.com/kubernetes-sigs/cluster-api-provider-openstack`
     - `git push upstream $VERSION`
-1. Run `make release` to build artifacts (the image is automatically built by CI)
+   This will cause the image to be automatically built by CI and pushed to the staging repository. As this only builds
+   the image, it only takes a few minutes.
 1. Follow the [image promotion process](https://github.com/kubernetes/k8s.io/tree/main/k8s.gcr.io#image-promoter) to promote the image from the staging repo to `k8s.gcr.io/capi-openstack`.
    The staging repository can be inspected at https://console.cloud.google.com/gcr/images/k8s-staging-capi-openstack/GLOBAL. Be
    sure to choose the top level `capi-openstack-controller`, which will provide the multi-arch manifest, rather than one for a specific architecture.
    The image build logs are available at [Cloud Build](https://console.cloud.google.com/cloud-build/builds?project=k8s-staging-capi-openstack).
    Add the new sha=>tag mapping to the [images.yaml](https://github.com/kubernetes/k8s.io/edit/main/k8s.gcr.io/images/k8s-staging-capi-openstack/images.yaml) (use the sha of the image with the corresponding tag). The PR to update the [images.yaml](https://github.com/kubernetes/k8s.io/edit/main/k8s.gcr.io/images/k8s-staging-capi-openstack/images.yaml) must be approved in the [OWNERS](https://github.com/kubernetes/k8s.io/blob/main/k8s.gcr.io/images/k8s-staging-capi-openstack/OWNERS) file and merged.
-1. Create a draft release in GitHub based on the tag created above
+1. Run `make release` to build artifacts to be attached to the GitHub release.
 1. Generate and finalize the release notes and add them to the draft release:
```

Comment:
Can't add the release notes to the draft release that is only created in the next step.

```diff
- 1. Generate and finalize the release notes and add them to the draft release:
+ 1. Generate and finalize the release notes and save them for the next step:
```